app/tests/reader_studies_tests/test_views.py

```diff
@@ -454,8 +425,9 @@ def test_display_set_update(client):
         },
         user=user,
         method=client.post,
+        content_type="application/json",
     )
-    assert response.status_code == 302
+    assert response.status_code == 200
```

Comment:
Why are we no longer redirecting on creation? https://en.wikipedia.org/wiki/Post/Redirect/Get

Reply:
In the latest implementation the reason for this is that the view we're calling here is an HTMX view. In order to redirect on success, we need to return an HttpResponse with the appropriate HX- headers. Django's HttpResponseRedirect unfortunately does not support these headers, and so we are stuck with HttpResponse, hence the 200 status code.

Reply:
So why not `HttpResponse` with the headers and status code 302?

Reply:
That would work. I thought I should better not change the status codes of these classes, but I see now that HttpResponse is a pretty generic response, unlike the specific ones like HttpResponseRedirect etc, so ok.